lib/sensors/src/accelerometer.rs

```diff
@@ -7,56 +7,75 @@ use hyped_io::i2c::{HypedI2c, I2cError};
 /// The accelerometer is configured to a sampling rate of 200Hz with high performance mode and continuous update.
 /// The full scale of the accelerometer (+-16g) is used.
 ///  
-/// The acceleration value for each axis is read from the sensor and converted to a floating point value in gs. 
-/// 
+/// The acceleration value for each axis is read from the sensor and converted to a floating point value in gs.
+///
 /// Data sheet: https://www.st.com/resource/en/datasheet/lis2ds12.pdf
 /// Application notes: https://www.st.com/resource/en/application_note/an4748-lis2ds12-alwayson-3axis-accelerometer-stmicroelectronics.pdf
-pub struct Accelerometer<T: HypedI2c> {
-    i2c: T,
+pub struct Accelerometer<'a, T: HypedI2c + 'a> {
+    i2c: &'a mut T,
     device_address: u8,
 }
 
-impl<T: HypedI2c> Accelerometer<T> {
+impl<'a, T: HypedI2c> Accelerometer<'a, T> {
     /// Create a new instance of the accelerometer and attempt to configure it
-    pub fn new(mut i2c: T, device_address: AccelerometerAddresses) -> Result<Self, AccelerometerError> {
+    pub fn new(
+        i2c: &'a mut T,
+        device_address: AccelerometerAddresses,
+    ) -> Result<Self, AccelerometerError> {
         let device_address = device_address as u8;
-        match i2c.write_byte_to_register(device_address, LIS2DS12_CTRL1_ADDRESS, LIS2DS12_CTRL1_VALUE) {
-            Err(e) => return Err(AccelerometerError::I2cError(e)),
-            _ => ()
-        };
-        match i2c.write_byte_to_register(device_address, LIS2DS12_CTRL2_ADDRESS, LIS2DS12_CTRL2_VALUE) {
-            Err(e) => return Err(AccelerometerError::I2cError(e)),
-            _ => ()
-        };
-        match i2c.write_byte_to_register(device_address, LIS2DS12_FIFO_CTRL_ADDRESS, LIS2DS12_FIFO_CTRL_VALUE) {
-            Err(e) => return Err(AccelerometerError::I2cError(e)),
-            _ => ()
-        };
-
+        if let Err(e) =
+            i2c.write_byte_to_register(device_address, LIS2DS12_CTRL1_ADDRESS, LIS2DS12_CTRL1_VALUE)
+        {
+            return Err(AccelerometerError::I2cError(e));
+        }
```

Comment:
maybe make a macro for this? idk if there's a nice way to generalise it across the different i2c users @davidbeechey?